### PR TITLE
Sanitising string on signature request pages

### PR DESCRIPTION
**Pull request by jpuri**
Sanitising string on signature request pages

Ref: https://app.zenhub.com/workspaces/confirmations-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-planning/306

### CODE REVIEW

File: ui/helpers/utils/util.js [hunk 1]
```diff
@@ -516,3 +517,20 @@ export function coinTypeToProtocolName(coinType) {
 export function isNullish(value) {
   return value === null || value === undefined;
 }
+
+/**
+ * The method escape RTL character in string
+ *
+ * @param {string} str
+ * @returns {string} escaped string
```
Comment (digiwand) [inline ui/helpers/utils/util.js:525]:
```suggestion
 * @returns {(string|*)} escaped string or original param value
```

`if (!str)`, then this function may return something other than a string

File: ui/components/app/signature-request/signature-request-data/signature-request-data.js [hunk 2]
```diff
@@ -19,61 +19,68 @@ import {
...
+                    className="signature-request-data__node__value__address"
+                  >
+                    <Address
+                      addressOnly
+                      checksummedRecipientAddress={toChecksumHexAddress(value)}
+                      recipientName={getAccountName(identities, value)}
+                    />
+                  </Typography>
+                ) : (
+                  value
```
Comment (digiwand) [inline ui/components/app/signature-request/signature-request-data/signature-request-data.js:77]:
```suggestion
                  `${value}`
```

I think we should add the string conversation back. 

If the value is an object, we may get the error `Uncaught Error: Objects are not valid as a React child` here; or if this is a boolean, the value may not print.

File: ui/components/app/signature-request-original/signature-request-original.component.js [hunk 3]
```diff
@@ -177,8 +177,12 @@ export default class SignatureRequestOriginal extends Component {
                 className="request-signature__row"
                 key={`request-signature-row-${index}`}
               >
-                <div className="request-signature__row-title">{`${name}:`}</div>
-                <div className="request-signature__row-value">{value}</div>
+                <div className="request-signature__row-title">{`${sanitizeString(
+                  name,
+                )}:`}</div>
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/signature-request-original/signature-request-original.component.js]: Thanks a lot @digiwand for bringing this up, I updated it to `{sanitizeString(`${name}:`)}`

File: ui/helpers/utils/util.js [hunk 4]
```diff
@@ -524,3 +525,20 @@ export function coinTypeToProtocolName(coinType) {
...
   return value === null || value === undefined;
 }
+
+/**
+ * The method escape RTL character in string
+ *
+ * @param {string} str
+ * @returns {(string|*)} escaped string or original param value
+ */
+export const sanitizeString = (str) => {
```
Comment (digiwand) [inline ui/helpers/utils/util.js:535]: I suppose the `str` param could be type any (*) now and maybe we want to rename the parameter to `value`. 

That, or we could allow this function to convert the value by renaming to function, e.g. `stringifyAndSanitize`, or adding an optional param, `stringify = true`

Comment (jpuri) [reply ui/helpers/utils/util.js]: Good point @digiwand : I renamed arg to `value`.

File: ui/helpers/utils/util.js [hunk 5]
```diff
@@ -540,9 +538,27 @@ export function isNullish(value) {
...
   );
 
   return pathMetadata?.name ?? null;
 }
 ///: END:ONLY_INCLUDE_IN
+
+/**
+ * The method escape RTL character in string
+ *
+ * @param {any} value
```
Comment (digiwand) [inline ui/helpers/utils/util.js:552]: maybe 
```suggestion
 * @param {*} value
```
ref: https://jsdoc.app/tags-param.html

**Review by digiwand — APPROVED** [review]
Nice fix!

**Review by segun — APPROVED** [review]

File: ui/helpers/utils/util.js (hunk 5 above)
Comment (jpuri) [reply ui/helpers/utils/util.js:552]: Good catch @digiwand , I will add this in a following PR.

Comment (jpuri) [reply ui/helpers/utils/util.js:552]: Done in this PR https://github.com/MetaMask/metamask-extension/pull/17537

**Comment by seaona** [comment]
I see it fixed!! Great job @jpuri !!

**Pull request merged**